docs/backends/Google.md

```diff
@@ -262,3 +262,37 @@ This filesystem has two required configuration options:
   documentation](https://www.ncbi.nlm.nih.gov/books/NBK63512/#Download.are_downloaded_files_encrypted)
   for more information on obtaining your NGC.  The `ngc` value provided above
   is the sample credential file.
+
+### Virtual Private Network
+
+In PAPIV2, to run your jobs in private network add the `virtual-private-cloud` stanza in papiv2 section in `backend` stanza.
+
+```
+backend {
+  ...
+  providers {
+  	...
+  	PapiV2 {
+  	  actor-factory = "cromwell.backend.google.pipelines.v2alpha1.PipelinesApiLifecycleActorFactory"
+  	  config {
+  		...
+  		virtual-private-cloud {
+  	          network-label-key = "my-private-network"
+  	          auth = "reference-to-auth-scheme"
+  	        }
+  	    ...
+  	  }  
+      }
+  }
+}
+```
+
+
+The `network-label-key` should reference the key in the label in your project, whose value is the name of your private network. And the `auth` should reference to an auth scheme in `google` stanza which should be used to request Google Cloud to get the project's metadata. For example, if your `virtual-private-cloud` config looks like the one above, and one of the labels in your project is
+
+```
+"my-private-network" = "vpc-network"
+```
+
+Cromwell will get labels from the project's metadata and look for a label whose key is `my-private-network`. Then, it will use the value of the label, which is `vpc-network` here, as the name of private network and run the jobs on this network.
+If the network key is not present in project's metadata, Cromwell will fail the job with error similar to `Unable to extract labels containing network information. Project metadata does not have network label key <network-key>`.
```

Comment:
Actually on reflection I'm not sure whether this is the desired behavior - if we have one value in our configuration we might knock out all pre-existing projects which don't have that value set up. Or projects which don't want to bother with subnetworks.

Maybe @helgridly would be able to comment on whether this is the right default behavior?

Reply:
I talked to @helgridly, and it seems that is definitely not the desired behavior. The behavior should be `if unable to find network key in the project metadata, fallback to running on default network`. I will make the necessary changes. Thank you for bringing this up!

Reply:
Chiming in to confirm: what Saloni said. Check your conf; if no key in the conf, use default networks always. If key in the conf, check the project labels. If key in the project label, use that network/subnet; else use default network.

Reply:
@cjllanwarne I have updated this logic to fall back to running the job on default network if `network-key` is not present in project's metadata.